c10/util/SmallVector.h

```diff
@@ -1339,7 +1342,8 @@ class /* LLVM_GSL_OWNER */ SmallVector : public SmallVectorImpl<T>,
     return *this;
   }
 
-  SmallVector(SmallVector&& RHS) : SmallVectorImpl<T>(N) {
+  SmallVector(SmallVector&& RHS) noexcept(std::is_nothrow_move_assignable_v<T>)
```

Comment:
Need to double check whether or not the move assignments operators actually call move assignments on their contents or just move construction.